### ISSUE TITLE
[transform_filter] - condition not working

### ISSUE THREAD

**Issue by rpalloni** [altair-viz/altair#710]
I do not understand why the condition in transform_filter is not working and have not found any explanation around (including documentation).
1) The problem seems to be in FilterTransoform method.

2) another issue: where and how am I supposed to put/write the '**background**' to change the default transparent?

```python
import altair.vegalite.v2 as alt
import pandas as pd

stocks = pd.read_csv(data.stocks.url)
type(stocks)
stocks.head()
stocks.symbol
stocks['symbol'] = stocks['symbol'].astype('category') # forced to be a factor
stocks.symbol

base = alt.Chart(stocks).encode(
    x='date:T',
    y='price:Q',
    color='symbol:N'
).transform_filter(
    stocks.symbol == 'GOOG' # problem is here
)
base.mark_line()
```

**Comment by eitanlees** [comment]
When doing data transformation in Altair often you will need to use [Altair Expressions](https://altair-viz.github.io/user_guide/transform.html#altair-expressions)

Try changing `stocks.symbol == 'GOOG'` to `alt.datum.symbol == 'GOOG'`

**Comment by eitanlees** [comment]
If you only want to make the visualization you can avoid pandas all together. 

```python
import altair as alt
from vega_datasets import data

stocks = data.stocks.url

base = alt.Chart(stocks).encode(
    x='date:T',
    y='price:Q',
    color='symbol:N'
).transform_filter(
    alt.datum.symbol == 'GOOG' 
)
base.mark_line()
```

**Comment by rpalloni** [comment]
This pic also explains **2**. 
Where should I add instructions for a white background of the image (and not transparent)?

**Comment by eitanlees** [comment]
I noticed that in the `vegify` function it has a line about `vegalite.v1`. I wonder if you are using an older version of Vega-Lite. I think Altair is supposed to work for Vega-Lite 2.0 and higher. 

What IDE are you using?

As far as the white background, I am not sure. 

@jakevdp might be able to help

**Comment by rpalloni** [comment]
The filter is not working regardless the vegify.
Anyway I'm using Atom + Hydrogen + Jupyter kernel and for vegify I followed the hint here https://github.com/nteract/hydrogen/issues/560
Notice that using vegalite.v2 in def does not produce output then

**Comment by jakevdp** [comment]
I don't know what ``vegify`` is... that's unrelated to Altair, and this is the first I've ever heard of it, so I can't help with that.

Regarding ``transform_filter()``: the condition should be a string containing a [Vega Expression](https://vega.github.io/vega/docs/expressions/) string, which looks like ``datum.symbol === 'GOOG'``.  Here ``datum`` is not a new source of data, it is the way that filter expressions refer to data elements within Vega and Vega-Lite.

Passing it a Pandas boolean mask is not supported.

**Comment by jakevdp** [comment]
Oh, noticed your other question about backgrounds. If you want to configure the background, you can use

```
Chart().configure(
    background='white'
)
```

**Comment by jakevdp** [comment]
Another thing (I should really read questions more carefully before answering :smile:)

You're rendering your plot with Altair version 1; this doesn't support everything that Altair version 2 supports, and so filter transforms, etc. will not work properly.

nteract does not have Vega-Lite 2 support yet, and so I would not recommend it as a frontend for Altair at this time.

**Comment by rgbkrk** [comment]
@jakevdp - I just checked and vega v3 and vega-lite v2 are both supported in nteract desktop:

![screen shot 2018-04-09 at 1 06 28 pm](https://user-images.githubusercontent.com/836375/38520285-37426658-3bf7-11e8-9be4-c448cc0574fd.png)

Hydrogen needs to upgrade to the latest of the display area. nteract desktop is totally fine and has been since [nteract v0.6.0](https://github.com/nteract/nteract/releases/tag/v0.6.0), which was released at the end of January. We're now on v0.8.4 so any system that allows the auto-upgrades should be all set.
